portal/src/SwitchTabBlockerDialog.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext, useMemo } from "react";
+import {
+  IDialogContentProps,
+  Dialog,
+  DialogType,
+  DialogFooter,
+  PrimaryButton,
+  DefaultButton,
+  IDialogProps,
+  IButtonProps,
+} from "@fluentui/react";
+import { Context, FormattedMessage } from "@oursky/react-messageformat";
+
+export interface SwitchTabBlockerDialogProps extends IDialogProps {
+  onDialogConfirm?: IButtonProps["onClick"];
+  onDialogDismiss?: IButtonProps["onClick"];
+}
+
+const SwitchTabBlockerDialog: React.FC<SwitchTabBlockerDialogProps> = function SwitchTabBlockerDialog(props) {
```

Comment:
This component does not use any special hooks. NavigationBlockerDialog is special hooks + UI. Can we refactor NavigationBlockerDialog to use SwitchTabBlockerDialog in its implementation? Then this component can simply be renamed to BlockerDialog.

Reply:
Refactored.